gr-utils/python/blocktool/cli/parseheader.py

```diff
@@ -0,0 +1,301 @@
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+
+import os
+import json
+import logging
+import yaml
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except:
+    from yaml import Loader, Dumper
+
+import click
+
+from ..cli.base import BlockToolException, run_blocktool
+from ..core.parseheader import BlockHeaderParser
+from ..core import Constants
+
+LOGGER = logging.getLogger(__name__)
+
+## setup dumper for dumping OrderedDict ##
+_MAPPING_TAG = yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG
+
+
+def dict_representer(dumper, data):
+    """ Representer to represent special OrderedDict """
+    return dumper.represent_dict(data.items())
+
+
+def dict_constructor(loader, node):
+    """ Construct an OrderedDict for dumping """
+    return OrderedDict(loader.construct_pairs(node))
+
+
+Dumper.add_representer(OrderedDict, dict_representer)
+Loader.add_constructor(_MAPPING_TAG, dict_constructor)
+
+
+@click.command('parseheader',
+               short_help='Generate the parsed output for the header file or directory in a specified format')
+@click.argument('file-path', nargs=1)
+@click.option('-yaml', '--yaml-confirm', is_flag=True,
+              help='If given, a YAML *file* alongside printed JSON response will be generated')
+@click.option('-c', '--blocktool-comments', is_flag=True,
+              help='blocktool helper comments will be added in the header file')
+@click.option('-f', '--file-confirm', is_flag=True,
+              help='If given, file with default name in both JSON and YAML format will be generated')
+def cli(**kwargs):
+    """
+    \b
+    Get parsed output for a header file from GNU Radio module
+    """
+    kwargs['modtool'] = False
+    if os.path.isfile(kwargs['file_path']):
+        self = BlockHeaderParser(**kwargs)
+        self.cli = True
+        self.json_confirm = True
+        run_blocktool(self)
+        click.secho('Header file: {}'.format(self.filename), fg='green')
+        json_generator(self, **kwargs)
+        if kwargs['yaml_confirm']:
+            self.yaml_confirm = True
+            yaml_generator(self)
+    elif os.path.isdir(kwargs['file_path']):
+        parse_directory(**kwargs)
+    else:
+        raise BlockToolException('Invalid file or directory path.')
+
+
+def json_generator(self, **kwargs):
+    """
+    Generate JSON file for the block header
+    """
+    header = self.filename.split('.')[0]
+    block = self.modname.split('-')[-1]
+    if kwargs['file_confirm']:
+        click.secho('Successfully generated {}_{}.json'.format(
+            block, header), fg='green')
+        json_file = os.path.join('.', block+'_'+header + '.json')
+        with open(json_file, 'w') as _file:
+            json.dump(self.parsed_data, _file, indent=4)
+    else:
+        print(json.dumps(self.parsed_data, indent=4))
+        click.secho('Successfully parsed header file {}'.format(
+            self.filename), fg='green')
+
+
+def yaml_generator(self):
```

Comment:
this seems to be the blocktool to grc-yaml code THAT BELONGS IN MODTOOL
the --yaml option in the cli should just dump the info in yaml format instead of json - as was discussed numerous times